src/mui/field/ReferenceManyField.js

```diff
@@ -30,31 +42,26 @@ export class ReferenceManyField extends Component {
     render() {
         const { resource, reference, referenceRecords, children, basePath } = this.props;
         if (React.Children.count(children) !== 1) {
-            throw new Error('<ReferenceManyField> only accepts a single child');
+            throw new Error('<ReferenceManyField> only accepts a single child (like <Datagrid>)');
         }
         if (typeof referenceRecords === 'undefined') {
             return <LinearProgress style={{ marginTop: '1em' }} />;
         }
         const referenceBasePath = basePath.replace(resource, reference); // FIXME obviously very weak
-        return (
-            <div style={{ display: 'flex', flexWrap: 'wrap' }}>
-                {Object.keys(referenceRecords).map(index =>
-                    React.cloneElement(children, {
-                        key: index,
-                        record: referenceRecords[index],
-                        resource: reference,
-                        basePath: referenceBasePath,
-                    })
-                )}
-            </div>
-        );
+        return React.cloneElement(children, {
+            resource: reference,
+            ids: Object.keys(referenceRecords).map(id => parseInt(id, 10)),
```

Comment:
http://stackoverflow.com/questions/17106681/parseint-vs-unary-plus-when-to-use-which/17106702#17106702